libs/platform/experience/composition/composition-components.controller.ts

```diff
@@ -74,11 +74,16 @@ export class CompositionComponentsController implements ReactiveController {
         if (rules.hide) {
           return of(null);
         }
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return this.tokenResolver.resolveToken(rules.hideByRule!).pipe(
-          //hidden by default
-          startWith(true),
-          map((value) => (value ? null : component))
+
+        const hideRules =
```

Comment:
I believe this new `OR` logic should be a part of DefaultTokenService

Reply:
Why it should be part of DefaultTokenService? This service's responsibility is resolving tokens, not rules. Rules could have different "shapes" that's why I think the operations with rules should proceed in the controller layer and then pass it to TokenService.

Reply:
Because not only composition uses token resolver, if some other component or service requires similar logic of resolving, should we duplicate the code then? Or add additional operator like AND, need to update each component then?

I believe that TokenService should be able to accept any string, even with conditions. In result it's purpose to receive string and return resolved value